Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp

```diff
@@ -233,8 +233,10 @@ CaptureSourceOrError UserMediaCaptureManager::DisplayFactory::createDisplayCaptu
     if (m_shouldCaptureInGPUProcess)
         return CaptureSourceOrError { "Display capture in GPUProcess is not implemented"_s };
 #endif
-    if (m_shouldCaptureInGPUProcess)
-        m_manager.m_remoteCaptureSampleManager.setVideoFrameObjectHeapProxy(&WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy());
+    if (m_shouldCaptureInGPUProcess) {
+        RefPtr videoFrameObjectHeapProxy = &WebProcess::singleton().ensureGPUProcessConnection().videoFrameObjectHeapProxy();
+        m_manager.m_remoteCaptureSampleManager.setVideoFrameObjectHeapProxy(videoFrameObjectHeapProxy.get());
```

Comment:
Please pass `WTFMove(videoFrameObjectHeapProxy)` here and update `setVideoFrameObjectHeapProxy()` to take in a `RefPtr<RemoteVideoFrameObjectHeapProxy>&&` and `WTFMove()` inside it.

This will avoid refcounting churn.